dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageCheckerMain.java

```diff
@@ -56,19 +57,23 @@ public static void main(String[] arguments) throws IOException, RepositoryExcept
       if (linkageCheckerArguments.hasInput()) { 
         // This is non-empty if a BOM or artifacts are specified in the argument
         ImmutableList<Artifact> artifacts = linkageCheckerArguments.getArtifacts();
-    
+
         // When JAR files are specified in the argument, artifacts are empty.
-        ImmutableList<Path> inputClassPath;
-        ImmutableSet<Path> entryPointJars;
+        ImmutableList<AnnotatedJar> inputClassPath;
+        ImmutableSet<AnnotatedJar> entryPointJars;
         List<ArtifactProblem> artifactProblems = new ArrayList<>();
         // classPathResult is kept null if JAR files are specified in the argument
         ClassPathResult classPathResult = null;
     
         // FIXME the if here isn't reachable
         if (artifacts.isEmpty()) {
-          // When JAR files are passed as arguments, classPathResult is null, because there is no need
+          // When JAR files are passed as arguments, classPathResult is null, because there is no
+          // need
           // to resolve Maven dependencies.
-          inputClassPath = linkageCheckerArguments.getJarFiles();
+          inputClassPath =
```

Comment:
This is more complex than the code we had before. :-(

Reply:
Good point. ` linkageCheckerArguments.getJarFiles()` should return `AnnotatedJar`.